### DISCUSSION TITLE
Windows Package Manager 1.6.1573-preview

### DISCUSSION THREAD

**Discussion by ryfu-msft** [microsoft/winget-cli#3318] [Releases]
This is the first development build after the Windows Package Manager 1.5 build for Windows 10 (1809+) and Windows 11. This build will be released to Windows Insider Dev builds and Windows Package Manager Insiders. 

Experimental features are enabled in this release. Run `winget features` to see which experimental features are enabled or disabled. Add the following to your settings (`winget settings`) file to enable the experimental features such as `WinGet configuration`:

```JSON
    "experimentalFeatures": {
	  "dependencies": true,
	  "directMSI": true,
          "configuration": true,
	  "windowsFeature": true,
    },
```

Windows Package Manager now includes [Winget configuration](https://learn.microsoft.com/windows/package-manager/configuration/), which automatically handles the setup and configuration requirements for an ideal development environment on your Windows machine. WinGet configuration file helps with installing and managing software packages, applications, programming languages, frameworks, tools, or settings necessary for a project. 

Check out our [session](https://aka.ms/BRK280H) at Microsoft Build to learn how to get your machine to a ready-to-code state.

A prerelease version of the [Microsoft.WinGet.Client PowerShell module](https://www.powershellgallery.com/packages/Microsoft.WinGet.Client/0.2.1) has been published to the PowerShell Gallery and will no longer be included as a release asset. To install the latest version of the PowerShell module, run the following command in PowerShell 7+. 

```pwsh
Install-Module -Name Microsoft.WinGet.Client
```

## What's Changed
* Fix spelling mistake in german by @Speedfighter in https://github.com/microsoft/winget-cli/pull/3250
* Add icon extraction to installer metadata collection by @yao-msft in https://github.com/microsoft/winget-cli/pull/3235
* Add initial version of manifest 1.5 by @yao-msft in https://github.com/microsoft/winget-cli/pull/3255
* Improve correlation for optimized search in rest source and improve general correlation by downloading manifests by @yao-msft in https://github.com/microsoft/winget-cli/pull/3252
* Microsoft.WinGet.Configuration messages by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3242
* Disable warnings in external projects by @florelis in https://github.com/microsoft/winget-cli/pull/3256
* Add User Setting controlling Path Tokenization by @Trenly in https://github.com/microsoft/winget-cli/pull/3209
* Configuration cancellation support by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3244
* Update build projects by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3264
* Fix new warning C26495 from wil header in VS 17.6 by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3266
* Add Microsoft.WinGet.Configuration module to ReleaseStatic configuration by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3262
* Add WinGet configuration by @denelon in https://github.com/microsoft/winget-cli/pull/3277
* Make --uninstall-previous argument stable by @Trenly in https://github.com/microsoft/winget-cli/pull/3282
* Add icons entries to yaml manifest, rest interface and com interface by @yao-msft in https://github.com/microsoft/winget-cli/pull/3273
* Treat warning not as error for CS8785 by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3301
* Move HasData check to default installation location check only by @yao-msft in https://github.com/microsoft/winget-cli/pull/3302
* Don't read past the end of the HTTP stream by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3300
* Update localization from build 2306.01001 by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3316
* Make pinning a stable feature by @florelis in https://github.com/microsoft/winget-cli/pull/3315
* Bump version from 1.5 to 1.6 by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3317

## New Contributors
* @Speedfighter made their first contribution in https://github.com/microsoft/winget-cli/pull/3250

**Full Changelog**: https://github.com/microsoft/winget-cli/compare/v1.5.1361-preview...v1.6.1573-preview

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/winget-cli/releases/tag/v1.6.1573-preview'>Windows Package Manager 1.6.1573-preview</a>.</em>

**Comment by felipecrs** [comment]
Is it normal that this `msixbundle` is that big? Previous releases were much smaller.

**Reply by denelon** [reply]
We're including .NET and PowerShell hosting while WinGet configuration is in preview. We've got some other future work to help deal with the size "better".

**Reply by felipecrs** [reply]
Got it. Thank you!

**Reply by JohnMcPMS** [reply]
Future work is going to make the bundle file a bit smaller, but still a lot larger than pre-configuration (so instead of 30 times as large, maybe only 20 times...).  One way to reduce the overall download is to install from the URL directly, like:
```
PS> Add-AppxPackage https://github.com/microsoft/winget-cli/releases/download/v1.6.1573-preview/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle
OR
Win+R> ms-appinstaller:?source=https://github.com/microsoft/winget-cli/releases/download/v1.6.1573-preview/Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle
```
or if we did the work to support channels so we could put our own package in winget-pkgs with both channels...

**Reply by felipecrs** [reply]
It's good to know that I can supply an URL to Add-AppxPackage directly. 😅

But nevertheless, it's all good. I don't actually mind with packages getting bigger so long they have good reasons to (which seems to be the case). The intention of my comment was just to check if you were not bundling something by mistake into the package.

**Comment by Hermholtz** [comment]
> Future work is going to make the bundle file a bit smaller, but still a lot larger than pre-configuration (so instead of 30 times as large, maybe only 20 times...). 

I hope we're gonna get 100 times more features than RPM has with its mere 4 MBs in download size.
That's just ridiculous. Maybe also add Node.js, a headless browser, a few Electron apps, and a Linux virtual machine and make it 5 GB in size. It's so Microsoft.

**Reply by felipecrs** [reply]
At the end of the day, it's very unlikely that this size difference will make any noticeable difference for the end user.

It would be certainly better if we could have the same features set in a smaller size, but well:

1. We already have games taking 150GB of space for no apparent good reason. We don't let to play these games, so I don't think this is big deal.
2. I think it's better to focus on adding new features, then optimize it later. WinGet is relatively new.
3. At least it's open source. I believe any contribution to optimize the package size would be appreciated.

Anyway, that's just what I think.
